core/src/main/java/org/elasticsearch/cluster/routing/allocation/RerouteExplanation.java

```diff
@@ -50,15 +60,23 @@ public Decision decisions() {
         return this.decisions;
     }
 
+    public List<String> warnings() { return this.warnings; }
+
+    public boolean hasWarnings() {
+        return warnings().size() > 0;
+    }
+
     public static RerouteExplanation readFrom(StreamInput in) throws IOException {
         AllocationCommand command = in.readNamedWriteable(AllocationCommand.class);
         Decision decisions = Decision.readFrom(in);
-        return new RerouteExplanation(command, decisions);
+        List<String> warnings = in.readList(StreamInput::readString);
+        return new RerouteExplanation(command, decisions, warnings);
     }
```

Comment:
I noticed that `RerouteExplanation` and `RoutingExplanations` don't implement `Writeable` but still do this serialization to StreamOutput; wasn't sure if it made sense to change it in this PR (or at all)